### PR TITLE
[refactor]: persistance of sorting and page in table

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Fixes: https://github.com/SigNoz/signoz/issues/4153

Loom: https://www.loom.com/share/b5a39e196b424645af31f73f8706c8de?sid=69bb0c09-ff76-4946-9d96-fec9bf08be69

### CODE REVIEW

File: frontend/src/container/ListAlertRules/ListAlert.tsx [hunk 1]
```diff
@@ -128,6 +158,13 @@ function ListAlert({ allAlertRules, refetch }: ListAlertProps): JSX.Element {
 		}
 	};
 
+	const handleSearch = useDebouncedFn((e: unknown) => {
+		setSearchString((e as React.BaseSyntheticEvent).target.value);
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/ListAlert.tsx:162]: nit pick, can assign e.target.value to a variable and use in both lines.

File: frontend/src/container/ListAlertRules/ListAlert.tsx [hunk 2]
```diff
@@ -142,6 +179,10 @@ function ListAlert({ allAlertRules, refetch }: ListAlertProps): JSX.Element {
 				return prev - next;
 			},
 			render: DateComponent,
+			sortOrder:
+				sortedInfo.columnKey === DynamicColumnsKey.CreatedAt
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/ListAlert.tsx:183]: Do we need safety checks for sortedInfo.columnKey and order ?

File: frontend/src/container/ListAlertRules/ListAlert.tsx [hunk 3]
```diff
@@ -271,26 +319,36 @@ function ListAlert({ allAlertRules, refetch }: ListAlertProps): JSX.Element {
...
 		<>
-			<ButtonContainer>
-				<TextToolTip
-					{...{
-						text: `More details on how to create alerts`,
-						url: 'https://signoz.io/docs/userguide/alerts-management/',
-					}}
+			<SearchContainer>
+				<Search
+					placeholder="Search with Alert Name, Severity and Labels"
```
Comment (YounixM) [inline frontend/src/container/ListAlertRules/ListAlert.tsx:324]: `Search by`

**Review by YounixM — DISMISSED** [review]

File: frontend/src/container/ListAlertRules/ListAlert.tsx (hunk 1 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ListAlertRules/ListAlert.tsx:162]: Done.

File: frontend/src/container/ListAlertRules/ListAlert.tsx (hunk 2 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ListAlertRules/ListAlert.tsx:183]: No, an empty string has been passed as a column key by default.

File: frontend/src/container/ListAlertRules/ListAlert.tsx (hunk 3 above)
Comment (Rajat-Dabade) [reply frontend/src/container/ListAlertRules/ListAlert.tsx:324]: Done.

**Review by srikanthccv — APPROVED** [review]
Tested persistance of

- Search text
- Search text filter results after the poll 
- Status, alert name, severity order key
- Pagination

**Review by YounixM — APPROVED** [review]

**Pull request merged**